paparazzi-gradle-plugin/src/test/java/app/cash/paparazzi/gradle/PaparazziPluginTest.kt

```diff
@@ -188,6 +188,58 @@ class PaparazziPluginTest {
     assertThat(resourceFileContents.any { it.contains("release") }).isFalse()
   }
 
+  @Test
+  fun prepareResourcesCachingDependencyVersionBump() {
+    val fixtureRoot = File("src/test/projects/prepare-resources-task-caching")
+    val buildDir = fixtureRoot.resolve("build").registerForDeletionOnExit()
+    val buildGradleFile = File(fixtureRoot, "build.gradle")
+    val buildGradleFileNew = File(fixtureRoot, "build.gradle.bak")
+    val buildGradleFileOriginalBackup = File(fixtureRoot, "build.gradle.bak.original")
+    fixtureRoot.resolve("build-cache").registerForDeletionOnExit()
+
+    val firstRun = gradleRunner
+      .withArguments("testDebug", "--build-cache", "--stacktrace")
+      .forwardOutput()
+      .runFixture(fixtureRoot) { build() }
+
+    with(firstRun.task(":preparePaparazziDebugResources")) {
+      assertThat(this).isNotNull()
+      assertThat(this!!.outcome).isNotEqualTo(FROM_CACHE)
+    }
+
+    var resourcesFile = File(fixtureRoot, "build/intermediates/paparazzi/debug/resources.json")
+    assertThat(resourcesFile.exists()).isTrue()
+    var resourceFileContents = resourcesFile.readLines()
+    assertThat(resourceFileContents.any { it.contains("core-1.10.0") }).isTrue()
+
+    // delete now (regardless of future cleanup)
+    buildDir.deleteRecursively()
+
+    // Bump core dependency version by renaming build gradle
+    assertThat(buildGradleFile.renameTo(buildGradleFileOriginalBackup)).isTrue()
+    assertThat(buildGradleFileNew.renameTo(buildGradleFile)).isTrue()
+
+    try {
+      val secondRun = gradleRunner
+        .withArguments("testDebug", "--build-cache", "--stacktrace")
+        .forwardOutput()
+        .runFixture(fixtureRoot) { build() }
+
+      with(secondRun.task(":preparePaparazziDebugResources")) {
+        assertThat(this).isNotNull()
+        assertThat(this!!.outcome).isNotEqualTo(FROM_CACHE)
+      }
+
+      resourcesFile = File(fixtureRoot, "build/intermediates/paparazzi/debug/resources.json")
+      assertThat(resourcesFile.exists()).isTrue()
+      resourceFileContents = resourcesFile.readLines()
+      assertThat(resourceFileContents.any { it.contains("core-1.10.1") }).isTrue()
+    } finally {
+      assertThat(buildGradleFile.renameTo(buildGradleFileNew)).isTrue()
+      assertThat(buildGradleFileOriginalBackup.renameTo(buildGradleFile)).isTrue()
+    }
+  }
```

Comment:
this looks like what `verifyResourcesUpdatedWhenExternalDependencyChanges` intended to test, but maybe the versioning was the key part?  can we fix/reuse that test per these changes, or is there a noteworthy difference to justify another test?